src/prover/proof.service.ts

```
@@ -158,13 +152,30 @@ export class ProofService implements OnModuleInit {
 
   /**
    * Gets the minimum date that a proof can be generated for a given chain id.
-   * @param chainID  the chain id
+   * @param prover
    * @returns
    */
   getProofMinimumDate(prover: ProofType): Date {
     return addSeconds(new Date(), this.getProofMinimumDurationSeconds(prover))
   }
 
+  /**
+   * Get ProofType from string
+   * @param proof
+   * @private
+   */
+  private getProofTypeFromString(proof: string): ProofType {
+    switch (proof) {
+      case 'storage':
+        return PROOF_STORAGE
+      case 'Hyperlane':
+      case 'Metalayer':
+        return PROOF_HYPERLANE
+      default:
```

for storage prover, you should just throw here instead of returning a type, that way AI wont complain about dead code and other people wont need to remember that storage is disabled

---

Done ✅  https://github.com/eco/eco-solver/pull/137/commits/8df66c1f7df01efcd44b7c5b162fabacd73daa98

We should probably remove the storage prover logic. I'll open a PR to clean up some of the death logic.